lib/lhm/connection.rb

```diff
@@ -8,36 +9,74 @@ class Connection < SimpleDelegator
     alias connection __getobj__
     alias connection= __setobj__
 
-    def initialize(connection:, default_log_prefix: nil, retry_options: {})
+    def initialize(connection:, default_log_prefix: nil, options: {}, retry_config: {})
       @default_log_prefix = default_log_prefix
-      @retry_options = retry_options || default_retry_config
+      @retry_options = retry_config || default_retry_config
       @sql_retry = Lhm::SqlRetry.new(
         connection,
-        retry_options,
+        options: retry_config,
+        reconnect_with_consistent_host: options[:reconnect_with_consistent_host] || false
       )
 
       # Creates delegation for the ActiveRecord Connection
       super(connection)
     end
 
-    def execute(query, retry_options = {})
-      exec_with_retries(:execute, query, retry_options)
+    def options=(options)
+      # If any other flags are added. Add the "processing" here
+      @sql_retry.reconnect_with_consistent_host = options[:reconnect_with_consistent_host] || false
     end
 
-    def update(query, retry_options = {})
-      exec_with_retries(:update, query, retry_options)
+    def execute(query, should_retry: false, retry_options: {})
+      if should_retry
+        exec_with_retries(:execute, query, retry_options)
+      else
+        exec(:execute, query)
+      end
     end
 
-    def select_value(query, retry_options = {})
-      exec_with_retries(:select_value, query, retry_options)
+    def update(query, should_retry: false, retry_options: {})
+      if should_retry
+        exec_with_retries(:update, query, retry_options)
+      else
+        exec(:update, query)
+      end
+    end
+
+    def select_value(query, should_retry: false, retry_options: {})
+      if should_retry
+        exec_with_retries(:select_value, query, retry_options)
+      else
+        exec(:select_value, query)
+      end
+    end
+
+    def select_values(query, should_retry: false, retry_options: {})
+      if should_retry
+        exec_with_retries(:select_values, query, retry_options)
+      else
+        exec(:select_values, query)
+      end
+    end
+
+    def select_one(query, should_retry: false, retry_options: {})
+      if should_retry
+        exec_with_retries(:select_one, query, retry_options)
+      else
+        exec(:select_one, query)
+      end
```

Comment:
Fine for now, it occured to me that AR allows for more than just queries to be passed in here (and possibly also bind arguments?), not that we use this feature..